tpot/tpot.py

```diff
@@ -979,31 +868,63 @@ def gradient_boosting(input_df, learning_rate, n_estimators, max_depth):
         if max_depth < 1:
             max_depth = None
 
-        # If there are no features left (i.e., only 'class', 'group', and 'guess' remain in the DF), then there is nothing to do
+
+        return TPOT._train_model_and_predict(input_df, GradientBoostingClassifier, learning_rate=learning_rate, n_estimators=n_estimators, max_depth=max_depth, random_state=42)
+    
+    @staticmethod
+    def _train_model_and_predict(input_df, model, **kwargs):
+        """Fits an arbitrary sklearn classifier model with a set of keyword parameters
+
+        Parameters
+        ----------
+        input_df: pandas.DataFrame {n_samples, n_features+['class', 'group', 'guess']}
+            Input DataFrame for fitting the k-neares
+        model: sklearn classifier
+            Input model to fit and predict on input_df
+        kwargs: unpacked parameters
+            Input parameters to pass to the model's constructor, does not need to be a dictionary
+
+        Returns
+        -------
+        input_df: pandas.DataFrame {n_samples, n_features+['guess', 'group', 'class', 'SyntheticFeature']}
+            Returns a modified input DataFrame with the guess column updated according to the classifier's predictions.
+            Also adds the classifiers's predictions as a 'SyntheticFeature' column.
+
+        """
+        #Validate input
+        #If there are no features left (i.e., only 'class', 'group', and 'guess' remain in the DF), then there is nothing to do
         if len(input_df.columns) == 3:
             return input_df
-        
+
         input_df = input_df.copy()
 
         training_features = input_df.loc[input_df['group'] == 'training'].drop(['class', 'group', 'guess'], axis=1).values
         training_classes = input_df.loc[input_df['group'] == 'training', 'class'].values
+       
+        # Try to seed the random_state parameter if the model accepts it.
+        try:
+            clf = model(random_state=42,**kwargs)
```

Comment:
I'm pondering if seeding the random state is even necessary. Maybe it's even harmful in the long run?

This isn't something we need to address here, but I've been considering exploring that more.